### PR TITLE
Adds support for OpenCLIP

### PR DESCRIPTION
**Pull request by LysandreJik**
Adds support for OpenCLIP as a library with a code snippet.

cc @rwightman

### CODE REVIEW

**Comment by osanseviero** [comment]
(this is cool!)

For usage of `hf_hub_download` in openclip, setting `library_name` and `library_version` is also nice :)

**Comment by LysandreJik** [comment]
Thanks for the feedback, I'll add that to the OpenCLIP PR!

**Comment by rwightman** [comment]
Looks good, 

@osanseviero you mean? https://github.com/mlfoundations/open_clip/blob/main/src/open_clip/pretrained.py#L14

**Comment by osanseviero** [comment]
Yes, maybe https://github.com/mlfoundations/open_clip/blob/main/src/open_clip/pretrained.py#L282 should also change but I'm not very familiar with the codebase

**Comment by rwightman** [comment]
> Yes, maybe https://github.com/mlfoundations/open_clip/blob/main/src/open_clip/pretrained.py#L282 should also change (which is the function that made me say the comment above). but I'm not very familiar with the codebase

It's using the fn I linked to above, it bound the lib/version args

**Comment by julien-c** [comment]
i don't think `library_name` and `library_version` are used in Elasticsearch stats currently (unless i'm mistaken?)

**Comment by osanseviero** [comment]
Yes, this does not impact download stats but this should allow us to do fine grained analysis at any point in the future if there's any interest in that. We have this in all the integrations.

**Comment by julien-c** [comment]
yes i was only commenting on @rwightman's question 
> if there are a decent number of existing downloads [based on this]

**Comment by rwightman** [comment]
we should probably line up the names across lib name and tags either way? but good to know re elasticsearch stats.

**Comment by LysandreJik** [comment]
Updated the code snippet to also feature the tokenizer and updated `openclip` to `open_clip` to be coherent.

Switching this to non-draft mode as https://github.com/mlfoundations/open_clip/pull/375 was merged.

File: js/src/lib/interfaces/Libraries.ts [hunk 1]
```diff
@@ -16,6 +16,7 @@ export enum ModelLibrary {
 	"keras"                  = "Keras",
 	"k2"                     = "K2",
 	"nemo"                   = "NeMo",
+	"open_clip"               = "OpenCLIP",
```
Comment (Pierrci) [inline js/src/lib/interfaces/Libraries.ts:19]:
```suggestion
	"open_clip"              = "OpenCLIP",
```
😃

**Review by Pierrci — COMMENTED** [review]
LGTM

File: js/src/lib/interfaces/Libraries.ts (hunk 1 above)
Comment (LysandreJik) [reply js/src/lib/interfaces/Libraries.ts:19]: oops, good catch! Thanks @Pierrci

**Review by coyotte508 — APPROVED** [review]

**Comment by LysandreJik** [comment]
Thanks for the reviews @Pierrci and @coyotte508!

**Pull request merged**